# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.h
# Commit: 22649272ffae
# Full Hash: 22649272ffae2ec3077c452cf5acce0326758623
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move DOM module loader into own file; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137531
# ==============================================================================

diff -r e3fe772947bf -r 22649272ffae dom/script/ScriptLoader.h
--- a/dom/script/ScriptLoader.h	Fri Feb 18 12:45:51 2022 +0000
+++ b/dom/script/ScriptLoader.h	Fri Feb 18 12:45:51 2022 +0000
@@ -23,7 +23,7 @@
 #include "mozilla/dom/LoadedScript.h"
 #include "mozilla/dom/JSExecutionContext.h"  // JSExecutionContext
 #include "mozilla/dom/ScriptLoadRequest.h"
-#include "ModuleLoaderBase.h"
+#include "ModuleLoader.h"
 #include "mozilla/MaybeOneOf.h"
 #include "mozilla/MozPromise.h"
 #include "ScriptKind.h"
@@ -711,52 +711,6 @@
   RefPtr<ScriptLoader> mLoader;
 };
 
-class ModuleLoader final : public ModuleLoaderBase {
- private:
-  virtual ~ModuleLoader();
-
- public:
-  explicit ModuleLoader(ScriptLoader* aLoader);
-
-  ScriptLoader* GetScriptLoader() {
-    return static_cast<ScriptLoader*>(mLoader.get());
-  }
-
-  // Methods that must be overwritten by an extending class
-  void EnsureModuleHooksInitialized() override;
-
-  /**
-   * Start a load for a module script URI.
-   * Sets up the necessary security flags before calling StartLoadInternal.
-   * Short-circuits if the module is already being loaded.
-   */
-  nsresult StartModuleLoad(ScriptLoadRequest* aRequest) override;
-
-  void ProcessLoadedModuleTree(ModuleLoadRequest* aRequest) override;
-
-  nsresult CompileOrFinishModuleScript(
-      JSContext* aCx, JS::Handle<JSObject*> aGlobal,
-      JS::CompileOptions& aOptions, ModuleLoadRequest* aRequest,
-      JS::MutableHandle<JSObject*> aModuleScript) override;
-
-  // Create a top-level module load request.
-  static already_AddRefed<ModuleLoadRequest> CreateTopLevel(
-      nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
-      const SRIMetadata& aIntegrity, nsIURI* aReferrer, ScriptLoader* aLoader,
-      DOMScriptLoadContext* aContext);
-
-  // Create a module load request for a static module import.
-  already_AddRefed<ModuleLoadRequest> CreateStaticImport(
-      nsIURI* aURI, ModuleLoadRequest* aParent) override;
-
-  // Create a module load request for dynamic module import.
-  static already_AddRefed<ModuleLoadRequest> CreateDynamicImport(
-      nsIURI* aURI, ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-      DOMScriptLoadContext* aContext, ScriptLoader* aLoader,
-      JS::Handle<JS::Value> aReferencingPrivate,
-      JS::Handle<JSString*> aSpecifier, JS::Handle<JSObject*> aPromise);
-};
-
 }  // namespace dom
 }  // namespace mozilla
 